### ISSUE TITLE
Vulkan: Permanent FPS drop and VRAM utilization increase after viewport is resized many times

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#81258]
### Godot version

4.2.dev3, 4.2.dev 541674d10 (could reproduce this in 4.0.beta too)

### System information

Fedora 38, GeForce RTX 4090 (NVIDIA 535.98)

### Issue description

https://github.com/Calinou/godot-photo-mode-demo exhibits permanently increased VRAM usage after settings are temporarily increased once (from 3.8 GB to 5.3 GB), and then twice (from 5.3 GB to 5.5 GB). Subsequent settings toggles don't affect VRAM usage significantly:

https://github.com/godotengine/godot/assets/180032/77b21bc5-010d-4a94-a538-737bb8827ae1

However, if I perform the same operation in 4K, a permanent performance drop occurs after VRAM is fully exhausted by the settings toggle. (This photo mode internally renders the viewport in 8K if the window size is 4K.)

https://github.com/godotengine/godot/assets/180032/a49fd143-d085-4353-ba8c-0c6d6ffd0666

VRAM exhaustion is supposed to destroy performance, but I think performance should go back to normal levels after VRAM usage goes to normal levels and previous settings are restored.

This seems to be made more evident when enabling Environment effects such as SDFGI and volumetric fog.

### Steps to reproduce

- Use a FPS monitor while project is running (built-in or external).
- Monitor VRAM utilization using an external program such as MangoHud, LibreHardwareMonitor or `nvidia-smi`.
- In a 3D project, resize the window or viewport's 3D buffer at run-time using `get_viewport().scaling_3d_scale`.

### Minimal reproduction project

https://github.com/Calinou/godot-photo-mode-demo

I can try to create a more minimal MRP if needed.

**Comment by Zireael07** [comment]
Suspicion in the linked issue: https://github.com/godotengine/godot/issues/60110#issuecomment-1685097195

TLDR buffers are created but not deleted
